app/src/main/kotlin/Reversi/ReversiState.kt

```diff
@@ -40,4 +40,36 @@ class ReversiState : Cloneable {
     public override fun clone(): ReversiState {
         return ReversiState(this)
     }
+
+    public override fun equals(other: Any?): Boolean {
+        if (this === other) {
```

Comment:
We are overriding the Kotlin primitive `.equals` with a referential equate operator. Could we instead use `===` while this is implemented to avoid the override?

Reply:
Oh nevermind, i see there are a bunch of addditional constraints on the `.equals` override. But nevertheless, I'd go for something like `equalsStrict` instead of overriding the primitive.

Reply:
We want to use Kotlin collections dictionary, why `equals` being overwritten.